### PR TITLE
Add support for STIG standard

### LINKED ISSUE #9558: Load STIG standard after ACOMMONS-11 is done
Next week, a new version of analyzer commons will be added via https://sonarsource.atlassian.net/browse/ACOMMONS-11 to enable the import of a new security standard.

As part of the SQ plugin hardening sprint, we should update Analyzer commons and call the metadata API.

~~Check the PR for adding support for ASVS  standard - should be the same thing as in https://github.com/SonarSource/sonar-dotnet/pull/5941~~
Edit by Mary: Functionality of this PR is no longer valid. We use the common functionality to add security standards by sonar-commons-io. Should be enough to update the package sonar-commons-io and add tests.

- [ ] Add a check for the SonarQube version (it will be shipped in 10.7 in September). On SonarCloud it's already supported.
- [ ] Tests are in [CSharpSonarRulesDefinitionTest ](https://github.com/SonarSource/sonar-dotnet/blob/9.30.0.95878/sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpSonarRulesDefinitionTest.java#L91).

### PR DESCRIPTION
**Pull request by Tim-Pohlmann**
Fixes #9558

### CODE REVIEW

File: sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpSonarRulesDefinitionTest.java [hunk 1]
```diff
@@ -36,13 +36,13 @@
...
-  private static final String SECURITY_HOTSPOT_RULE_KEY = "S5766";
-  private static final String VULNERABILITY_RULE_KEY = "S4426";
+  private static final String SECURITY_HOTSPOT_RULE_KEY = "S4502";
+  private static final String VULNERABILITY_RULE_KEY = "S2053";
   private static final String NO_TAGS_RULE_KEY = "S1048";
   private static final String SINGLE_PARAM_RULE_KEY = "S1200";
   private static final String MULTI_PARAM_RULE_KEY = "S110";
 
-  private static final SonarRuntime SONAR_RUNTIME = SonarRuntimeImpl.forSonarQube(Version.create(9, 9), SonarQubeSide.SCANNER, SonarEdition.COMMUNITY);
+  private static final SonarRuntime SONAR_RUNTIME = SonarRuntimeImpl.forSonarQube(Version.create(10, 10), SonarQubeSide.SCANNER, SonarEdition.COMMUNITY);
```
Comment (mary-georgiou-sonarsource) [inline sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpSonarRulesDefinitionTest.java:45]: I know this is cleaner - but the STIG are being raised only from SQ 10.7+ while OWASP from 9.9+ so I'd keep this tests seperate.

File: sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractRulesDefinitionTest.java [hunk 2]
```diff
@@ -37,6 +37,7 @@ public class AbstractRulesDefinitionTest {
 
   private static final String PCI_DSS_RULE_KEY = "S1115";
```
Comment (mary-georgiou-sonarsource) [inline sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractRulesDefinitionTest.java:38]: CAYC: this rule is deleted, and it's not C#. Maybe we can replace it with another one?

**Review by mary-georgiou-sonarsource — CHANGES_REQUESTED** [review]
Looks ok but IMO we should keep tests seperate.

File: sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractRulesDefinitionTest.java (hunk 2 above)
Comment (Tim-Pohlmann) [reply sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractRulesDefinitionTest.java:38]: I also stumbled upon this at first, but this is not actually S1115. The rule IDs here refer to some fake rules in [AbstractRulesDefinitionTest](https://github.com/SonarSource/sonar-dotnet/tree/87aff2914deef3bb95faafa0cd0d0e93b268937e/sonar-dotnet-shared-library/src/test/resources/AbstractRulesDefinitionTest).

File: sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpSonarRulesDefinitionTest.java (hunk 1 above)
Comment (Tim-Pohlmann) [reply sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpSonarRulesDefinitionTest.java:45]: `SONAR_RUNTIME` is used for all tests in this file. Only a few of them are related to security standards. I don't think using multiple runtime objects is worthwhile unless there are some breaking changes.

Comment (mary-georgiou-sonarsource) [reply sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpSonarRulesDefinitionTest.java:45]: Ok let's keep on single runtime - however I think we should split these tests to be on UT per standard.

**Review by mary-georgiou-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**